frontend/src/modules/WellCompletion/wellCompletionsDataAccessor.ts

```diff
@@ -0,0 +1,236 @@
+import {
+    WellCompletionDataSet_api,
+    WellCompletionData_api,
+    WellCompletionUnits_api,
+    WellCompletionWell_api,
+    WellCompletionZone_api,
+} from "@api";
+import { CompletionPlotData, PlotData, WellPlotData } from "@webviz/well-completions-plot";
+
+import { getRegexPredicate } from "./utils/stringUtils";
+
+/**
+ * Util methods for providing data result based on selected time aggregation
+ */
+export const TimeAggregations = {
+    None: (arr: number[]): number => arr[arr.length - 1],
+    Max: (arr: number[]): number => Math.max(...arr),
+    Average: (arr: number[]): number => arr.reduce((a, b) => a + b) / arr.length,
+};
+export type TimeAggregation = keyof typeof TimeAggregations;
+
+export class WellCompletionsDataAccessor {
+    /**
+     * This class is responsible for parsing and providing data for the well completions plot.
+     *
+     * It parses data structure received from backend and creates the data structure needed for the plot
+     * in WellCompletionsPlot-component.
+     *
+     * The data is parsed from the WellCompletionData_api object and methods are taken from the
+     * WellCompletions-component of webviz-subsurface-components package, and modified to fit.
+     */
+    private _dataSet: WellCompletionDataSet_api | undefined;
```

Comment:
It is a good practice to set intentionally not defined variables to null in order to make a clear separation from uncontrolled undefined variables.